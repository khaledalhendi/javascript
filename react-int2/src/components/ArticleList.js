import React from "react";
import Article from "./Article";

const ArticleList = props => {
  const { articles, articleActions } = props;
  //console.log("ArticleList");
  //console.log(articleActions);
  return (
    <div>
      {Object.values(articles).map(article => (
        <Article
          key={article.id}
          article={article}
          articleActions={articleActions}
        />
      ))}
    </div>
  );
};
export default ArticleList;
