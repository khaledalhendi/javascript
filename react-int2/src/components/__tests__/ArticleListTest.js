import React from "react";
import ArticleList from "../ArticleList";

import renderer from "react-test-renderer";

describe("ArticleList", () => {
  const testProps = {
    articles: { a: { id: "a" }, b: { id: "b" } },
    articleActions: {
      lookupAuthor: jest.fn(() => 0)
    }
  };

  it("renders correctly", () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
