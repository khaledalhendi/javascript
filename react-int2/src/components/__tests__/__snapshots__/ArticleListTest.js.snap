// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontwWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={undefined}
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontwWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={undefined}
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
</div>
`;
